core/src/main/scala/persistentscheduler/impl/PersistentScheduler.scala

```diff
@@ -7,30 +7,37 @@ import persistentscheduler.impl.PersistentScheduler._
 import persistentscheduler.scaladsl.SchedulerPersistence
 import persistentscheduler.{SchedulerSettings, TimedEvent}
 
-import scala.compat.java8.OptionConverters._
-import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
 
 private[persistentscheduler] object PersistentScheduler {
 
   sealed trait Result
-  case class Info(self: ActorRef)              extends Result
+
+  case class Info(self: ActorRef) extends Result
+
   case class SubscribedActorRef(ref: ActorRef) extends Result
-  case class Scheduled(event: TimedEvent)      extends Result
+
+  case class Scheduled(event: TimedEvent) extends Result
```

Comment:
`Scheduled` and `RemoveEventsByReference` aren't used anymore.

Also the `SubscribedActorRef` above probably will be useless as well because:

In the `subscribe` method of the `PersistentSchedulerExtension` you do a `.mapTo[Result]`, but in the `addSubscription` method you send back the `SubscribedActorRef` and not a Unit so I guess the extension would fail then to cast the response. So I think you can just send back a Unit instead of this result.